lab_1_keywords_tfidf/main.py

```diff
@@ -136,10 +91,16 @@ def get_top_n(frequencies: dict[str, Union[int, float]], top: int) -> Optional[l
 
     In case of corrupt input arguments, None is returned
     """
-    checking_dict = check_dict(frequencies, str, int, False) or check_dict(frequencies, str, float, False)
-    if not (checking_dict and check_positive_int(top)):
+
+    if not isinstance(frequencies, dict) or not frequencies or not isinstance(top, int) or isinstance(top,
+                                                                                                      bool) or top <= 0:
         return None
-    return sorted(frequencies.keys(), key=lambda key: frequencies[key], reverse=True)[:top]
+    if top > len(frequencies):
+        get_sorted_n = sorted(frequencies.items(), key=lambda x: x[1], reverse=True)
+    else:
+        get_sorted_n = sorted(frequencies.items(), key=lambda x: x[1], reverse=True)[:top]
+    n_top = list(dict(get_sorted_n))
```

Comment:
dictionaries are thought of as an unordered data type, so it is better not to convert something to the dict if order matters